compiler/src/dotty/tools/dotc/transform/Pickler.scala

```diff
@@ -137,11 +137,14 @@ class Pickler extends Phase {
   }
 
   private def testSame(unpickled: String, previous: String, cls: ClassSymbol)(using Context) =
-    if (previous != unpickled) {
+    import java.nio.charset.StandardCharsets.UTF_8
+    def normal(s: String) = new String(s.getBytes(UTF_8), UTF_8)
+    val unequal = unpickled.length() != previous.length() || normal(unpickled) != normal(previous)
+    if unequal then
```

Comment:
I didn't track down the encoding issue with unpickled as compilationUnit.toString. The behavior was that strings are unequals but array dump to txt files are the same.